nifi-commons/nifi-record/src/test/java/org/apache/nifi/serialization/record/ResultSetRecordSetTest.java

```diff
@@ -272,6 +278,76 @@ public void testCreateSchemaArrayThrowsNotSupportedException() throws SQLExcepti
         assertEquals(RecordFieldType.ARRAY.getArrayDataType(RecordFieldType.STRING.getDataType()), resultSchema.getField(0).getDataType());
     }
 
+    @Test
+    public void testCreateSchemaWithLogicalTypes() throws SQLException {
+        testCreateSchemaLogicalTypes(true);
+    }
+
+    @Test
+    public void testCreateSchemaNoLogicalTypes() throws SQLException {
+        testCreateSchemaLogicalTypes(false);
+    }
+
+    private void testCreateSchemaLogicalTypes(boolean useLogicalTypes) throws SQLException {
+        // GIVEN
+        Object[][] columns = new Object[][] {
```

Comment:
This structure seems difficult to maintain as it requires precise index references in other methods.  It seems like it would be better and easier to use one or or Maps with more precise types.  Another option might be a List of a custom objects declared in this test class.

Reply:
I agree. I eventually introduced a custom type (TestColumn) to avoid using Object[][]. Modifications are in the next commit. Thanks!